openvdb/Types.h

```diff
@@ -316,6 +316,70 @@ namespace internal {
 
 // Implementation details of TypeList
 
+struct NullType {};
+
+// Get a particular element
+
+// resolve if Idx is out-of-range
+template<typename ListT, size_t Idx, bool Valid> struct TSResolveElementImpl;
+template<typename... Ts, size_t Idx>
+struct TSResolveElementImpl<TypeList<Ts...>, Idx, true> {
+    using type = typename std::tuple_element<Idx, std::tuple<Ts...>>::type;
+};
+template<typename... Ts, size_t Idx>
+struct TSResolveElementImpl<TypeList<Ts...>, Idx, false> {
+    using type = NullType;
+};
+
+template<typename ListT, size_t Idx> struct TSGetElementImpl;
+
+template<typename... Ts, size_t Idx>
+struct TSGetElementImpl<TypeList<Ts...>, Idx> {
+private:
+    static constexpr bool Valid =
+        Idx < sizeof...(Ts) && sizeof...(Ts) > 0;
+public:
+    using type = typename TSResolveElementImpl<TypeList<Ts...>, Idx, Valid>::type;
+};
+
+
+// Check a type exists in a TypeList
+template <typename ListT, typename T, size_t Idx=0>
+struct TSHasTypeImpl;
+
+template <typename T>
+struct TSHasTypeImpl<TypeList<>, T> {
+    static constexpr bool Value = false;
+    static constexpr int64_t Index = -1;
+};
+
+template <typename U, typename T, typename... Ts, size_t Idx>
+struct TSHasTypeImpl<TypeList<U, Ts...>, T, Idx> :
+    TSHasTypeImpl<TypeList<Ts...>, T, Idx+1> {};
+
+template <typename T, typename... Ts, size_t Idx>
+struct TSHasTypeImpl<TypeList<T, Ts...>, T, Idx>
+{
+    static constexpr bool Value = true;
+    static constexpr int64_t Index = static_cast<int64_t>(Idx);
+};
+
+
+// // Remove any duplicate types from a TypeList
+// template <typename T, typename... Ts>
+// struct TSMakeUniqueImpl {
+//     using type = T;
+// };
+
+// template <typename... Ts, typename U, typename... Us>
+// struct TSMakeUniqueImpl<TypeList<Ts...>, U, Us...> {
+//     using type =
+//         typename std::conditional<(std::is_same<U, Ts>::value || ...),
+//            typename TSMakeUniqueImpl<TypeList<Ts...>, Us...>::type,
+//            typename TSMakeUniqueImpl<TypeList<Ts..., U>, Us...>::type >::type;
+// };
```

Comment:
`(...)` requires c++17 - we don't actually use `TypeList::Unique` anywhere so this is unnecessary. But a c++14 solution should be fairly straightforward.

Reply:
Fixed for c++14